php/Snoopy.class.inc

```diff
@@ -654,9 +658,17 @@ class Snoopy
 			$port = $this->port;
 		}
 		$this->status = 0;
-		if(function_exists("stream_socket_client") && !is_null($this->IP))
+		if(function_exists("stream_socket_client"))
 		{
-			$opts = array('socket' => array('bindto' => $this->IP.':0'));
+			$opts = array('socket' => array());
+			if(!is_null($this->IP))
+			{
+				$opts['socket']['bindto'] = $this->IP.':0';
+			}
+			elseif($this->preferIpv4)
```

Comment:
We can go `if($this->preferIpv4)` `$opts['socket']['bindto'] = '0:0';` here.
Then `else` `$opts['socket']['bindto'] = $this->IP.':0';`

Then check further upward: `if(function_exists("stream_socket_client") && !is_null($this->IP))`

Reply:
Wouldn't you want `$this->IP` to be trump all other settings? Since its a manually specified setting? Just a little confused on the branching order you are talking about here.